src/components/ScenarioParameters/components/ScenarioParametersInputs/GenericTable.js

```diff
@@ -470,6 +476,51 @@ export const GenericTable = ({
     }
   });
 
+  const onAddRow = useCallback(() => {
+    const newLine = ConfigUtils.createNewTableLine(parameterData.options.columns, parameterData.options.dateFormat);
+    const selectedLines = [
+      ...document.getElementById(`table-${parameterData.id}`).getElementsByClassName('ag-row-selected'),
+    ];
+    let newLineArray = parameter?.agGridRows;
+    if (newLineArray && selectedLines.length > 0) {
+      const lastSelectedLineIndex = selectedLines[selectedLines.length - 1].attributes['row-id'].value;
+      newLineArray.splice(parseInt(lastSelectedLineIndex) + 1, 0, newLine);
+    } else if (newLineArray) newLineArray.unshift(newLine);
+    else newLineArray = [newLine];
+    updateParameterValue({
+      status: UPLOAD_FILE_STATUS_KEY.READY_TO_UPLOAD,
+      tableDataStatus: TABLE_DATA_STATUS.READY,
+      errors: null,
+      uploadPreprocess: { content: _uploadPreprocess },
+      agGridRows: [...newLineArray],
+    });
+  }, [
+    _uploadPreprocess,
+    parameter?.agGridRows,
+    parameterData.id,
+    parameterData.options.columns,
+    parameterData.options.dateFormat,
+    updateParameterValue,
+  ]);
+  const onDeleteRow = useCallback(() => {
+    const selectedLines = document
+      .getElementById(`table-${parameterData.id}`)
+      .getElementsByClassName('ag-row-selected');
+    if (selectedLines.length > 0) {
+      const newRows = parameter.agGridRows;
+      for (let i = 0; selectedLines[i]; i++) {
+        newRows[selectedLines[i].attributes['row-id'].value] = undefined;
+      }
+      updateParameterValue({
+        status: UPLOAD_FILE_STATUS_KEY.READY_TO_UPLOAD,
+        tableDataStatus: TABLE_DATA_STATUS.READY,
+        errors: null,
+        uploadPreprocess: { content: _uploadPreprocess },
+        agGridRows: newRows.filter((row) => row),
+      });
+    }
+  }, [_uploadPreprocess, parameter.agGridRows, parameterData.id, updateParameterValue]);
```

Comment:
Using `document.getElementById`, and `getElementsByClassName` and elements ids will break every time ids or classes change in ag-grid, please use ag-grid API method `getSelectedRows` instead (see [doc](https://www.ag-grid.com/javascript-data-grid/grid-api/?ref=blog.ag-grid.com#reference-selection-getSelectedRows))